File: 9.palindrome-number/java/src/main/java/com/company/app/PalindromeNumber.java
package com.company.app;

public class PalindromeNumber
{
    public boolean _do(int num)
    {
      if (isNegative(num) || isZeroEnding(num)) {
        return false;
      }

      int converted = 0;

      while (num > converted) {
        converted = carryBy1Digit(converted) + getLastDigit(num);
        num = deleteLastDigit(num);
      }

      return (num == converted) || (num == (converted / 10));
    }

    private boolean isNegative(int num)
    {
      return num < 0;
    }

    private boolean isZeroEnding(int num)
    {
      return num % 10 == 0;
    }

    private int carryBy1Digit(int converted)
    {
      return converted * 10;
    }

    private int getLastDigit(int num)
    {
      return num % 10;
    }

    private int deleteLastDigit(int num)
    {
      return num / 10;
    }
}

File: 9.palindrome-number/golang/src/palindrome_number.go
package solution

func IsPalindrome(num int) bool {
	if isNegative(num) || isZeroEnding(num) {
		return false
	}

	converted := 0

	for num > converted {
		converted = carryBy1Digit(converted) + getLastDigit(converted)
		num = deleteLastDigit(num)
	}

	return (num == converted) || (num == (converted / 10))
}

func isNegative(num int) bool {
	return num < 0
}

func isZeroEnding(num int) bool {
	return (num%10 == 0)
}

func carryBy1Digit(converted int) int {
	return converted * 10
}

func getLastDigit(num int) int {
	return num % 10
}

func deleteLastDigit(num int) int {
	return num / 10
}
